### PR TITLE
Docs update

### PR DESCRIPTION
**Pull request by fufexan**
The much-needed docs update. Requested by Gytis, offered by me. Marking as draft as probably many things will change throughout this PR's life. Feel free to ping and correct me :slightly_smiling_face:

* README.md: fix typos and grammar, propose future names of library functions.
* examples: add a few comments and remove redundant channel declaration.

I believe we should bring back the `fully-featured` example, but now inspired from everyone's dotfiles in the Readme. Should follow the latest changes to `lib` and provide an easy starting point for newcomers.

Ideas are welcome.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-This flake exposes a library abstraction to *painlessly* generate nixos flake configurations.
+Flake-utils-plus exposes a library abstraction to *painlessly* generate NixOS flake configurations.
 
-The biggest design goal is to keep down the fluff. The library is meant to be easy to understand and use. It aims to be far simpler than frameworks such as devos (previously called nixflk).
+The biggest design goal is to keep down the fluff. The library is meant to be easy to understand and use. It aims to be far simpler than frameworks such as DevOS (previously called nixflk).
 
 # Features of flake #
 
-This flake provides two main features (visible from `flake.nix`):
+FUP provides two main features (visible from `flake.nix`):
```
Comment (Pacman99) [inline README.md:14]:
```suggestion
FUP provides a few main features (visible from `flake.nix`):
```

File: README.md [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
 
-- `nixosModules.saneFlakeDefaults` - Configures `nix.*` attributes. Generates `nix.nixPath`/`nix.registry` from flake `inputs`, sets `pkgs.nixUnstable` as the default also enables `ca-references` and `flakes`.
-- `lib.systemFlake { ... }` - Generates a system flake that may then be built.
-- `lib.exporter.modulesFromListExporter [ ./a.nix ./b.nix ]` - Generates modules attributes which looks like this `{ a = import ./a.nix; b = import ./b.nix; }`.
-- `lib.exporter.overlaysFromChannelsExporter channels` - Collects all overlays from channels and exports them as an appropriately namespaced attribute set. Users can instantiate with their nixpkgs version.
-- `lib.builder.packagesFromOverlayBuilderConstructor channels pkgs` - Similar to the overlay generator, but outputs them as packages, instead. Users can use your cache.
+- `nix.generateRegistryFromInputs` - Generates `nix.nixPath`/`nix.registry` from flake `inputs`.
+- `lib.mkFlake { ... }` - Generates a flake using FUP magic.
+- `lib.exportModules [ ./a.nix ./b.nix ]` - Generates modules attribute which looks like this `{ a = import ./a.nix; b = import ./b.nix; }`.
```
Comment (Pacman99) [inline README.md:18]:
```suggestion
- `lib.exportModules [ ./a.nix ./b.nix ]` - Generates module attribute which look like this `{ a = import ./a.nix; b = import ./b.nix; }`.
```

File: README.md [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
+- `lib.exportOverlays channels` - Exports all overlays from channels as an appropriately namespaced attribute set. Users can instantiate with their nixpkgs version.
+- `lib.outputsBuilder.packages channels pkgs` - Similar to the overlay generator, but outputs them as packages for the platforms defined in `meta.platforms`. Users can use your cache.
```
Comment (Pacman99) [inline README.md:20]:
```suggestion
- `lib.outputsBuilder.packages channels pkgs` - Similar to the overlay generator, but outputs them as packages for the platforms defined in `meta.platforms`. Unlike overlays, these packages are consistent across flakes allowing them to be cached.
```

**Review by Pacman99 — APPROVED** [review]
Great improvements!

File: README.md [hunk 4]
```diff
@@ -135,29 +139,33 @@ in flake-utils-plus.lib.systemFlake {
...
-
-  # Evaluates to `defaultPackage.<system>.neovim = <nixpkgs-channel-reference>.neovim`.
-  defaultPackageBuilder = channels: channels.nixpkgs.neovim;
 
-  # Evaluates to `apps.<system>.custom-neovim  = utils.lib.mkApp { drv = ...; exePath = ...; };`.
-  appsBuilder = channels: with channels.nixpkgs; {
-    custom-neovim = mkApp {
-      drv = fancy-neovim;
-      exePath = "/bin/nvim";
+  # THESE ARE GUESSES - pls review
```
Comment (gytis-ivaskevicius) [inline README.md:146]:
```suggestion
```

File: examples/exporters/flake.nix [hunk 5]
```diff
@@ -14,16 +14,20 @@
...
-      channels.nixpkgs.input = nixpkgs;
-
-      hosts.Morty.modules = with self.nixosModules; [
-        Morty
+      # explicitly add overlaysBuilder
+      channels.nixpkgs.overlaysBuilder = arg: [
+        (final: prev: { inherit coreutils; })
       ];
 
+      # propagates to channels.nixpkgs.overlaysBuilder
```
Comment (gytis-ivaskevicius) [inline examples/exporters/flake.nix:22]:
```suggestion
      # Channel specific overlays. Overlays `coreutils` from `unstable` channel.
      channels.nixpkgs.overlaysBuilder = channels: [
        (final: prev: { inherit (channels.unstable) coreutils; })
      ];

      # Propagates to channels.<name>.overlaysBuilder
```

File: examples/minimal-multichannel/flake.nix [hunk 6]
```diff
@@ -13,8 +13,8 @@
       inherit self inputs;
 
       # Channel definitions.
-      channels.unstable.input = unstable;
-      channels.nixpkgs.input = nixpkgs;
+      # Channels are automatically generated from nixpkgs inputs
+      # e.g the inputs which contain `legacyPackages`
```
Comment (gytis-ivaskevicius) [inline examples/minimal-multichannel/flake.nix:17]:
```suggestion
      # e.g the inputs which contain `legacyPackages` attribute are used.
```

**Review by gytis-ivaskevicius — APPROVED** [review]
Good stuff

File: README.md [hunk 7]
```diff
@@ -29,11 +29,14 @@ This flake provides two main features (visible from `flake.nix`):
 
 # How to use this flake #
 
-Example flake with all available attributes can be found [Here](https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/master/examples/fully-featured/flake.nix). (WARNING: Quite overwhelming)
+* [Example flake with all available attributes](https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/master/examples/fully-featured/flake.nix)
+TODO: we should add it back.
```
[the remark this answers is not in the document]
Comment (fufexan) [reply README.md:33]: Alright then.

File: README.md [hunk 8]
```diff
@@ -1,23 +1,23 @@
...
-- `lib.exporter.overlaysFromChannelsExporter channels` - Collects all overlays from channels and exports them as an appropriately namespaced attribute set. Users can instantiate with their nixpkgs version.
-- `lib.builder.packagesFromOverlayBuilderConstructor channels pkgs` - Similar to the overlay generator, but outputs them as packages, instead. Users can use your cache.
+- `nix.generateRegistryFromInputs` - Generates `nix.nixPath`/`nix.registry` from flake `inputs`.
+- `lib.mkFlake { ... }` - Generates a flake using FUP magic.
+- `lib.exportModules [ ./a.nix ./b.nix ]` - Generates module attribute which look like this `{ a = import ./a.nix; b = import ./b.nix; }`.
+- `lib.exportOverlays channels` - Exports all overlays from channels as an appropriately namespaced attribute set. Users can instantiate with their nixpkgs version.
+- `lib.outputsBuilder.packages channels pkgs` - Similar to the overlay generator, but outputs them as packages for the platforms defined in `meta.platforms`. Unlike overlays, these packages are consistent across flakes allowing them to be cached.
```
[the remark this answers is not in the document]
Comment (fufexan) [reply README.md:20]: Do you mean something like this?
```suggestion
- [`lib.exportModules [ ./a.nix ./b.nix ]`](./lib/exportModules.nix) - Generates module attribute which look like this `{ a = import ./a.nix; b = import ./b.nix; }`.
- [`lib.exportOverlays channels`](./lib/exportOverlays.nix) - Exports all overlays from channels as an appropriately namespaced attribute set. Users can instantiate with their nixpkgs version.
- [`lib.outputsBuilder.packages channels pkgs`](./lib/outputsBuilder.nix) - Similar to the overlay generator, but outputs them as packages for the platforms defined in `meta.platforms`. Unlike overlays, these packages are consistent across flakes allowing them to be cached.
```

**Comment by fufexan** [comment]
I think we should move the `# Documentation as code` part out of the README. Perhaps move it to `docs/all-options.nix`?

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (gytis-ivaskevicius) [reply README.md:20]: yeah, something like that. Tho I don't think that URL part will work the way you did it

**Pull request merged**